modules/k8s/secret.go

```diff
@@ -26,3 +30,23 @@ func GetSecretE(t testing.TestingT, options *KubectlOptions, secretName string)
 	}
 	return clientset.CoreV1().Secrets(options.Namespace).Get(context.Background(), secretName, metav1.GetOptions{})
 }
+
+// WaitUntilSecretAvailable waits until the secret is present on the cluster.
```

Comment:
NIT: Update this comment to provide the context you mentioned (letsencrypt) as an example use case for this method.